src/Microsoft.OData.Core/ODataMessageReader.cs

```diff
@@ -169,6 +169,16 @@ public ODataMessageReader(IODataResponseMessage responseMessage, ODataMessageRea
             this.model = model ?? GetModel(this.container);
             this.edmTypeResolver = new EdmTypeReaderResolver(this.model, this.settings.ClientCustomTypeResolver);
 
+            // Whether null values were omitted in the response, false by default.
+            this.settings.NullValuesOmitted = false;

```

Comment:
Setting the false as the default value, see comment in the line above.

---
In reply to: [198055075](https://github.com/OData/odata.net/pull/1106#discussion_r198055075) [](ancestors = 198055075)

Reply:
in this case, you are *always* setting the value to false, and then *possibly* setting it back to true.  Why not just set it to true if the preference apply matches, otherwise set it to false, as in:

            if (string.Equals(responseMessage.PreferenceAppliedHeader().OmitValues, ODataConstants.OmitValuesNulls,
                StringComparison.OrdinalIgnoreCase))
            {
                // If the Preference-Applied header's omit-values parameter is present in the response, its value should
                // be applied to the reader's setting.
                this.settings.NullValuesOmitted = true;
            }
            else
            {
                this.settings.NullValuesOmitted = false;
            }

---
In reply to: [209783953](https://github.com/OData/odata.net/pull/1106#discussion_r209783953) [](ancestors = 209783953,198055075)

Reply:
These two ways are mathematically equivalent. I still slightly incline to initializing it to false first in order to emphasize the fact that preference header omit-values=nulls is required to be set explicitly.

---
In reply to: [213509847](https://github.com/OData/odata.net/pull/1106#discussion_r213509847) [](ancestors = 213509847,209783953,198055075)

Reply:
I still prefer setting it once to twice as it feels cleaner, but I'm not going to force it.

---
In reply to: [213861831](https://github.com/OData/odata.net/pull/1106#discussion_r213861831) [](ancestors = 213861831,213509847,209783953,198055075)